java_src/foam/nanos/logger/NanoLogger.java

```diff
@@ -0,0 +1,89 @@
+package foam.nanos.logger;
+
+import foam.core.*;
+import foam.nanos.NanoService;
+
+import java.text.SimpleDateFormat;
+import java.sql.Timestamp;
+import java.io.IOException;
+import java.util.logging.*;
+
+public class NanoLogger extends ContextAwareSupport implements NanoService {
+  private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy.MM.dd.HH.mm.ss");
+  static Logger logger;
+
+  public void start() {
+    logger = Logger.getAnonymousLogger();
+    logger.setUseParentHandlers(false);
+    try {
+      Handler handler = new FileHandler("nano.log");
+      handler.setFormatter(new CustomFormatter());
+      logger.addHandler(handler);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private static class CustomFormatter extends Formatter {
+    long prevTime;
+    String prevTimestamp;
+    @Override
+    public String format(LogRecord record) {
+      if (prevTime / 1000 != System.currentTimeMillis() / 1000) {
+        prevTime = System.currentTimeMillis();
+        prevTimestamp = sdf.format(new Timestamp(prevTime));
+      }
+
+      StringBuffer sb = new StringBuffer();
+      int level = record.getLevel().intValue();
+      String msg = record.getMessage();
+
+      sb.append("[");
+      sb.append(prevTimestamp);
+      sb.append("] ");
+      sb.append("[");
+
+      if (level == 1000) {
```

Comment:
See https://github.com/foam-framework/foam/wiki/FOAM2-Style-Guidelines

This line should be:
if ( lev == 1000 ) {